rest/replicator_test.go

```diff
@@ -4657,14 +4662,15 @@ func TestLocalWinsConflictResolution(t *testing.T) {
 			remoteMutation: newRevisionState(4, "c", true, 0),
 			expectedResult: newRevisionState(5, "b", false, 0),
 		},
-		{
-			// simpleMutation includes an attachment prior to conflict, validates it's preserved
-			name:           "localAttachment",
-			initialState:   newRevisionState(3, "a", false, 2),
-			localMutation:  newRevisionState(4, "b", false, 0),
-			remoteMutation: newRevisionState(4, "c", false, 0),
-			expectedResult: newRevisionState(5, "b", false, 3), // revpos==3 here because the revision isn't replicated until rev 3
-		},
+		// TODO: Enable this test once CBG-1590 is implemented.
+		// {
+		// 	// simpleMutation includes an attachment prior to conflict, validates it's preserved
+		// 	name:           "localAttachment",
+		// 	initialState:   newRevisionState(3, "a", false, 2),
+		// 	localMutation:  newRevisionState(4, "b", false, 0),
+		// 	remoteMutation: newRevisionState(4, "c", false, 0),
+		// 	expectedResult: newRevisionState(5, "b", false, 3), // revpos==3 here because the revision isn't replicated until rev 3
+		// },
```

Comment:
Can you share the specifics of why these conflict resolution tests are failing under this PR and require CBG-1590, but are passing on master?  I would have expected at most an additional GetAttachment call, but nothing that would break replication.

Reply:
Sure Adam, the scenario is like this:

1. Create two SG instances rt1 and rt2 and start pull replication from rt2 to rt1.
2. Create initial revision of the document with an attachment on rt2
{"_attachments":{"initialAtt.txt":{"data":"aW5pdGlhbA=="}},"channels":["alice"]}
3. Wait for the document revision to get replicated to rt1.
4. Stop the replication process on rt1 and update the document by removing the attachment to create a local conflicting revision.
 {"source":"rt1","channels":["alice"]}
5. Create the remote conflicting revision by updating the doc on rt2 but keep the attachment stub.
 {"source":"rt2","_attachments":{"initialAtt.txt":{"stub":true,"revpos":1}},"channels":["alice"]}
6. Starts the replicator to trigger conflict resolution to merge both attachments in the conflict.

When rt1 receives the rev message, it tries to download the attachments. Subsequently rt1 identifies that the attachment is already uploaded and asks rt2 to prove it has the attachment without sending it. At that point, it panics and fails to prove the existence of the attachment — The issue is that in handleProveAttachment,  we're alway trying to grab the attachment by using the legacy format key, but the attachment is stored in the new storage format.